app/webpacker/components/RegistrationsV2/Register/index.jsx

```diff
@@ -53,22 +58,36 @@ function Register({
     },
   });
 
+  const onTimerEnd = useCallback(() => {
+    setTimerEnded(true);
+  }, [setTimerEnded]);
+
+  if (isFetching) {
+    return <Loading />;
+  }
+
+  if (userCanPreRegister(userInfo, competitionInfo) || competitionInfo['registration_currently_open?'] || timerEnded) {
```

Comment:
Lots of calculations inside one `if` statement, not happy about that

Reply:
should be better when I move userCanPreRegister to Ruby?